src/utils/setupNetwork.ts

```diff
@@ -20,6 +22,27 @@ export const setupNetwork = async (chainId: number) => {
           ],
         })
       }
+
+      if (chainId === ChainId.POLYGON) {
+        const config = NETWORK_CONFIGS[ChainId.POLYGON]
+        await provider.request({
+          method: 'wallet_addEthereumChain',
+          params: [
+            {
+              chainId: `0x${chainId.toString(16)}`,
+              chainName: config.name,
+              nativeCurrency: {
+                name: config.name,
+                symbol: config.symbol,
+                decimals: 18,
+              },
```

Comment:
Maybe this could be added to `NETWORK_CONFIG` and just take all the data from there.